### PR TITLE
PRD : Multiple Address : Ability to return Org Created date and Approved date

### PR DESCRIPTION
**Pull request by shivajilokare**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ### PRD : Multiple Address : Ability to return Org Created date and Approved date

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/request/OrganisationCreationRequest.java [hunk 1]
```diff
@@ -38,6 +38,10 @@
     @NotNull
     private List<ContactInformationCreationRequest> contactInformation;
 
+    private final String dateReceived;
+
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/request/OrganisationCreationRequest.java:42]: These changes are not required as there are not new fields is coming up in request

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/response/OrganisationEntityResponse.java [hunk 2]
```diff
@@ -91,5 +97,11 @@ private void getOrganisationEntityResponse(
                     .map(ContactInformationResponseWithDxAddress::new)
                     .collect(toList());
         }
+
+        this.dateReceived = organisation.getCreated().toString();
+
+        this.dateApproved = organisation.getDateApproved().toString();
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/response/OrganisationEntityResponse.java:103]: null pointer exception if date approved is null, make sure that if the date is null then needs to show this json field will null value.

May be check if toString is really required, try to keep fields as LocalDateTime

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java [hunk 3]
```diff
@@ -380,6 +381,7 @@ public OrganisationResponse updateOrganisation(
         if (isNotEmpty(savedOrganisation.getPaymentAccounts())
                 && organisationCreationRequest.getStatus().equals("ACTIVE")) {
             updatePaymentAccounts(savedOrganisation.getPaymentAccounts());
+            organisation.setDateApproved(LocalDateTime.now());
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java:384]: This is not right place to set Org date approved , Org is already persisted couple of lines above will not take any effect. Also need to make sure if there is Org approved request then only need to persists approveDate.
SuperController.updateOrganisationById is right place to add dateapproved field

**Review by shreedhar-hmcts — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java (hunk 3 above)
Comment (shivajilokare) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java]: resolved

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/request/OrganisationCreationRequest.java (hunk 1 above)
Comment (shivajilokare) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/request/OrganisationCreationRequest.java]: resolved

**Review by sahitya-desireddy — APPROVED** [review]

**Review by nayeemshaik-hmcts — APPROVED** [review]

**Review by sahitya-desireddy — APPROVED** [review]

**Review by nayeemshaik-hmcts — APPROVED** [review]
Approved

**Pull request merged**